### PR TITLE
feat: add support for nearcore 1.37 changes

### PR DESCRIPTION
**Pull request by gtsonevv**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/near/near-api-js/blob/master/CONTRIBUTING.md).
- [x] Commit messages follow the [conventional commits](https://www.conventionalcommits.org/) spec
- [ ] **If this is a code change**: I have written unit tests.
- [x] **If this changes code in a published package**: I have run `pnpm changeset` to create a `changeset` JSON document appropriate for this change.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

https://github.com/near/nearcore/blob/447572bb775a5954ca7a5a4b65c85b45aab9e499/chain/jsonrpc/CHANGELOG.md#changelog
https://github.com/near/nearcore/pull/9556
https://github.com/near/nearcore/pull/9648
https://github.com/near/nearcore/pull/9644
https://docs.google.com/document/d/1jGuXzBlMAGQENsUpy5x5Xd7huCLOd8k6tyMOEE41Rro/edit

### CODE REVIEW

File: packages/accounts/src/account.ts [hunk 1]
```diff
@@ -37,6 +36,7 @@ import {
 import BN from 'bn.js';
 
 import { Connection } from './connection';
+import { TxOutcome } from '@near-js/types/lib/provider/response';
```
Comment (vikinatora) [inline packages/accounts/src/account.ts]: Why isn't this imported like this?
```suggestion
import { TxOutcome } from '@near-js/types';
```

File: packages/providers/src/json-rpc-provider.ts [hunk 2]
```diff
@@ -80,26 +81,37 @@ export class JsonRpcProvider extends Provider {
...
+    /**
+     * Sends a signed transaction to the RPC
+     *
+     * @param signedTransaction The signed transaction being sent
+     * @param waitUntil
+     */
+    async sendTx(signedTransaction: SignedTransaction, waitUntil: TxExecutionStatus): Promise<TxOutcome> {
+        const bytes = encodeTransaction(signedTransaction);
+        return this.sendJsonRpc('send_tx', { signed_transaction: Buffer.from(bytes).toString('base64'), wait_until: waitUntil });
+    }
```
Comment (vikinatora) [inline packages/providers/src/json-rpc-provider.ts]: `sendTx` may not be the best name. The existing ones are:
- `sendTransaction` which wait for the transaction to be finalized
- `sendTransactionAsync` which doesn't wait for the transaction to be finalized.

I think that `sendTransactionUntil` could be a better name

File: packages/providers/src/json-rpc-provider.ts [hunk 3]
```diff
@@ -80,26 +81,37 @@ export class JsonRpcProvider extends Provider {
...
      * See [docs for more info](https://docs.near.org/docs/develop/front-end/rpc#send-transaction-async)
      * @param signedTransaction The signed transaction being sent
-     * @returns {Promise<FinalExecutionOutcome>}
+     * @returns {Promise<TxOutcome>}
      */
-    async sendTransactionAsync(signedTransaction: SignedTransaction): Promise<FinalExecutionOutcome> {
+    async sendTransactionAsync(signedTransaction: SignedTransaction): Promise<TxOutcome> {
         const bytes = encodeTransaction(signedTransaction);
-        return this.sendJsonRpc('broadcast_tx_async', [Buffer.from(bytes).toString('base64')]);
+        return this.sendJsonRpc('broadcast_tx_async', { signed_transaction: Buffer.from(bytes).toString('base64'), wait_until: 'None' });
```
Comment (vikinatora) [inline packages/providers/src/json-rpc-provider.ts]: From [here](https://github.com/near/nearcore/blob/447572bb775a5954ca7a5a4b65c85b45aab9e499/chain/jsonrpc/CHANGELOG.md#023): 
```
Added send_tx method which gives configurable execution guarantees options and potentially replaces existing broadcast_tx_async, broadcast_tx_commit
```
Won't it be better if these function just call `sendTx` with the proper `wait_until` parameter?

**Review by vikinatora — CHANGES_REQUESTED** [review]

File: packages/accounts/src/account.ts (hunk 1 above)
Comment (vikinatora) [reply packages/accounts/src/account.ts]: This applies to all imports of `TxOutcome`

Comment (gtsonevv) [reply packages/accounts/src/account.ts]: it was automatically imported, forgot to export `TxOutcome` from `@near-js/types`, that's why it's `@near-js/types/lib/provider/response`. Will fix it.

File: packages/providers/src/json-rpc-provider.ts (hunk 2 above)
Comment (gtsonevv) [reply packages/providers/src/json-rpc-provider.ts]: yes, `sendTransactionUntil` is better

**Pull request closed without merging**